### PR TITLE
ENGAGEUI-3105-chart-drillin-keyboard-accessible

### PR DESCRIPTION
**Pull request by KalaRaja**
added keyboard event in bars of column chart to make it keyboard accessible.

### CODE REVIEW

File: addon/components/column-chart/component.js [hunk 1]
```diff
@@ -286,6 +286,13 @@ export default BaseChartComponent.extend({
             return;
         }
 
+        chart.selectAll('rect.bar').each(function () {
+            const bar = d3.select(this);
+            if (bar.attr('height') !== '0') {
+                bar.attr('tabindex', 0);
+            }
+        });
```
Comment (Jbcampbe) [inline addon/components/column-chart/component.js]: maybe we can add this attribute in a place where we are already selecting all of the bars?

Comment (KalaRaja) [reply addon/components/column-chart/component.js]: Done.

File: addon/components/base-chart-component.js [hunk 2]
```diff
@@ -81,6 +81,17 @@ export default Component.extend({
             this.onClick(d);
         });
 
+        // when user navigates using tab key and uses enter button to drill-in the chart
+        const onClick = this.onClick;
+        this.get('chart').selectAll(this.elementToApplyTipSelector).each(function () {
```
Comment (shanebruggeman) [inline addon/components/base-chart-component.js]: Could remove `const onClick` above in favor of preserving `this` with an es6 function here
```suggestion
        this.get('chart').selectAll(this.elementToApplyTipSelector).each(() => {
```

Comment (shanebruggeman) [reply addon/components/base-chart-component.js]: And doing `this.onClick(...)` below

Comment (KalaRaja) [reply addon/components/base-chart-component.js]: in the same @each, i'm using `const bar = d3.select(this);` 'this' here is the context of the d3 selection and not of the entire component.

Comment (KalaRaja) [reply addon/components/base-chart-component.js]: Screen Shot 2019-11-26 at 1 51 12 PM

**Pull request merged**